src/lib/features/project/project-service.ts

```diff
@@ -558,8 +564,23 @@ export default class ProjectService {
             auditUser,
         );
 
+        const allTokens = await this.apiTokenStore.getAll();
+        const projectTokens = allTokens.filter(
+            (token) =>
+                (token.projects &&
+                    token.projects.length === 1 &&
+                    token.projects[0] === id) ||
+                token.project === id,
+        );
```

Comment:
This is essentially looking for tokens that only have a single project, and that project has id `id`, right?

Not important, but is there a nicer way to do it? I think even with 24k tokens it should be fine, but how about something like:

```ts
const tokens = await this.apiTokenStore.getTokensForProject(id);
const toDelete = tokens.filter(token => token.project || token.projects.length === 1)
```

Or something like that? 🤷🏼 Not important, though. Oh, I also don't know if we have a method like that, soooo

Reply:
I couldn't find `this.apiTokenStore.getTokensForProject`, I guess we could add this method though

Reply:
Yeah, it was just an idea I had (and def not a blocker). I think that method might be useful, but if this is the only place we do that, then ... eh 🤷🏼